From bf0e9050a2344ed2c974b0a324144808070dc7a6 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:42 -0400
Subject: [ethernet] i40e: remove unused variable and memory allocation

Message-id: <1409658042-22689-162-git-send-email-sassmann@redhat.com>
Patchwork-id: 90358
O-Subject: [RHEL7.1 PATCH 161/281] i40e: remove unused variable and memory allocation
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shannon Nelson <shannon.nelson@intel.com>

This was a vestige of early driver development that no longer
has any actual use.

Change-ID: I95b5b19c4bbfaff8759197af671ebaf716cb6ab5
Signed-off-by: Shannon Nelson <shannon.nelson@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 3146ce3d4da6bd8f5541c551a84719069bc9166e)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e.h b/drivers/net/ethernet/intel/i40e/i40e.h
index ebeaf95..2af28fd 100644
--- a/drivers/net/ethernet/intel/i40e/i40e.h
+++ b/drivers/net/ethernet/intel/i40e/i40e.h
@@ -297,7 +297,6 @@ struct i40e_pf {
  u16 pf_seid;
  u16 main_vsi_seid;
  u16 mac_seid;
- struct i40e_aqc_get_switch_config_data *sw_config;
  struct kobject *switch_kobj;
 #ifdef CONFIG_DEBUG_FS
  struct dentry *i40e_dbg_pf;
diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 6e08b54..35e79e4 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -7990,15 +7990,6 @@ int i40e_fetch_switch_configuration(struct i40e_pf *pf, bool printconfig)
      "header: %d reported %d total\n",
      num_reported, num_total);
 
-  if (num_reported) {
-   int sz = sizeof(*sw_config) * num_reported;
-
-   kfree(pf->sw_config);
-   pf->sw_config = kzalloc(sz, GFP_KERNEL);
-   if (pf->sw_config)
-    memcpy(pf->sw_config, sw_config, sz);
-  }
-
   for (i = 0; i < num_reported; i++) {
    struct i40e_aqc_switch_config_element_resp *ele =
     &sw_config->element[i];
@@ -8718,7 +8709,6 @@ static void i40e_remove(struct pci_dev *pdev)
 
  kfree(pf->qp_pile);
  kfree(pf->irq_pile);
- kfree(pf->sw_config);
  kfree(pf->vsi);
 
  /* force a PF reset to clean anything leftover */
-- 
1.7.1